File: data/test_sample.c
// “Sample” C file — with EM dash
#include <stdio.h>

// Function with EN dash – and non-breaking hyphen ‑
void hello_world() {
    printf("Hello, world!\n"); // “Hello” with smart quotes
    printf("Hyphen-minus: -\n");
    printf("Non-breaking hyphen: ‑\n");
    printf("EN dash: –\n");
    printf("EM dash: —\n");
    printf("Zero-width space:\u200b\n"); // Should be removed
    printf("Trailing whitespace here...   \n");
}

int main() {
    hello_world();
    return 0;
} 

File: data/test_sample.py
# This is a sample Python file — with EM dash
# Here’s a function with smart quotes and EN dash – and non-breaking hyphen ‑
def hello_world():
    print("Hello, world!")  # “Hello” with smart quotes
    print('Goodbye — world!')  # EM dash
    print('Hyphen-minus: -')
    print('Non-breaking hyphen: ‑')
    print('EN dash: –')
    print('EM dash: —')
    print('Zero-width space:')  # Should be removed
    print('Trailing whitespace here...   ')

if __name__ == '__main__':
    hello_world() 